src/convertToObject.js

```diff
@@ -3,16 +3,35 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ *  (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ *  and values are the values of related CSS properties
+ *    (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const styles = sourceString.split(';');
+
+  const splitedStyles = styles.map(style => style.split(':'));
+
+  const trimedStyles = splitedStyles.map(property => {
+    return property.map(value => value.trim());
+  });
+
+  const filteredStyles = trimedStyles.filter(property => property[0] !== '');
+
+  const result = filteredStyles.reduce((acc, property) => {
+    return {
+      ...acc,
+      [property[0]]: property[1],
+    };
+  }, {});
```

Comment:
you can save 2 useless loops by moving these operations inside your reducer. And now you have only 1 iteration insead of 3

```suggestion
  const result = filteredStyles.reduce((acc, [property, value]) => {
    const trimmedProperty = property.trim();
    const trimmedValue = value.trim();
    
    if (trimmedProperty.length && trimmedValue .length) {
      acc[trimmedProperty] = trimmedValue ;
    }
    
    return acc;
  }, {});
```